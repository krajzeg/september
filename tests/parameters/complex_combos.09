x := 0

crazy := |normal, ?lazy, ?...lazy_sink, ?:::lazy_named| {
	print("X before execution:", x)
	print("Normal parameter:", normal)
	print("Lazy parameter:", lazy.resolve())
	print("Lazy sink contents:")
	for (x) in (lazy_sink) {
		print(x.resolve())
	}
	print("Lazy named:", lazy_named.named_operation.resolve())
	print("X after execution:", x)
}

print("Should print: 0 1 2 6 10 50 50.")
crazy(named_operation: x = x * 5, 1, x = x + 2, x = x * 3, x = x + 4)
