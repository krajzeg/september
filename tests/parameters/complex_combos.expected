Should print: 0 1 2 6 10 50 50.
X before execution: 0
Normal parameter: 1
Lazy parameter: 2
Lazy sink contents:
6
10
Lazy named: 50
X after execution: 50
